lib/board.js

```diff
@@ -66,17 +70,20 @@ Board.prototype.makeBoardClickable = function () {
     board.canvas.addEventListener("mousemove", displayMoves);
   }
 
+  function indexBasedOnTurn (canvas_x, canvas_y) {
+    if (board.currentTurn ("white")) {
+      return (Math.floor(canvas_y / 100) * 8) + Math.floor(canvas_x / 100);
```

Comment:
There is an opportunity here to abstract out a few additional functions/methods.